pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -305,18 +565,38 @@ func TestVolumeBinding(t *testing.T) {
 			}
 
 			t.Logf("Verify: check state after prefilter phase")
-			stateData, err := getStateData(state)
+			got, err := getStateData(state)
 			if err != nil {
 				t.Fatal(err)
 			}
-			if !reflect.DeepEqual(stateData, item.wantStateAfterPreFilter) {
-				t.Errorf("state got after prefilter does not match: %v, want: %v", stateData, item.wantStateAfterPreFilter)
+			stateCmpOpts := []cmp.Option{
+				cmp.AllowUnexported(stateData{}),
+				cmpopts.IgnoreFields(stateData{}, "Mutex"),
+			}
+			if diff := cmp.Diff(item.wantStateAfterPreFilter, got, stateCmpOpts...); diff != "" {
+				t.Errorf("state got after prefilter does not match (-want,+got):\n%s", diff)
 			}
 
 			t.Logf("Verify: call Filter and check status")
-			gotStatus := p.Filter(ctx, state, item.pod, nodeInfo)
-			if !reflect.DeepEqual(gotStatus, item.wantFilterStatus) {
-				t.Errorf("filter status does not match: %v, want: %v", gotStatus, item.wantFilterStatus)
+			for i, nodeInfo := range nodeInfos {
+				gotStatus := p.Filter(ctx, state, item.pod, nodeInfo)
+				if !reflect.DeepEqual(gotStatus, item.wantFilterStatus[i]) {
+					t.Errorf("filter status does not match for node %q, got: %v, want: %v", nodeInfo.Node().Name, gotStatus, item.wantFilterStatus)
+				}
+			}
+
+			if item.feature == features.VolumeCapacityPriority {
```

Comment:
I think we should verify Score() unconditionally, as the plugin was instantiated taking the featuregate into consideration at L519. In other words, if the feature gate is disabled, Score() is a no-op.

Reply:
updated